metaflow/plugins/kfp/kfp_cli.py

```diff
@@ -54,6 +54,7 @@ def step_init(obj, run_id, step_name, passed_in_split_indexes, task_id):
     )
 
 
+@parameters.add_custom_parameters(deploy_mode=True)
```

Comment:
What does `deploy_mode=True` mean and how come our setting is True while it is False for step functions?
See https://github.com/zillow/metaflow/blob/tz/kfp_step/metaflow/plugins/aws/step_functions/step_functions_cli.py#L268-L268

Reply:
The difference between our kfp `run` command and the step_function `trigger` command you pointed out is, that our `run` function includes both deployment and trigger, while the step_function `trigger` only triggers the pipeline for existing deployments.

The deploy time parameter wrapper is defined [here](https://github.com/Netflix/metaflow/blob/master/metaflow/parameters.py#L50) With `deploy_mode=True` deploy time parameter is evaluated, otherwise they are completely ignored and omitted from pipeline. Any parameter with default being a callable is considered a deploy-time parameter

As an example, consider a parameter called `deploy_host_name` whose default is set to `socket.gethostname`. With `deploy_mode=True` after deployment this parameter should contain the deploy host name, and you will not find this parameter when you set `deploy_mode=False`. When you trigger from a host different from deploy host, by setting `deploy_mode=False`, you will not pass `deploy_host_name="<trigger_host_name>"` by accident.

Reply:
May I resolve this thread?